script.js

```diff
@@ -1,8 +1,68 @@
-const input = document.getElementsByClassName("addInput")[0];
-const listContainer = document.getElementsByClassName("listContainer")[0];
+const input = document.querySelector(".addInput");
+const listContainer = document.querySelector(".listContainer");
+const completedListContainer = document.querySelector(
+  ".completedListContainer"
+);
+const allButton = document.querySelector(".allButton");
+const clearCompletedButton = document.querySelector(".clearCompleted");
+allButton.addEventListener("click", function () {
+  filterTasks("all");
+});
+const tasks = [];
 
-function createListItem(taskContent) {
+function renderTasks(filteredTasks) {
+  listContainer.innerHTML = "";
+
+  filteredTasks.forEach((task) => {
+    const listItem = createListItem(task.title, task.status);
+    listContainer.appendChild(listItem);
+  });
+}
+
+clearCompletedButton.addEventListener("click", function () {
+  clearAllTasks();
+});
+
+function clearAllTasks() {
+  tasks.length = 0;
+  renderTasks([]);
+}
+
+function filterTasks(status) {
+  const filteredTasks = tasks.filter((task) => {
+    if (status === "all") {
+      return task;
+    } else if (status === task.status) {
+      return task;
+    }
+  });
+
+  renderTasks(filteredTasks);
+}
+
+function createListItem(taskContent, status) {
```

Comment:
Remove one createListItem function
The two createListItem functions can be merged for simplicity and bug prevention.